tests/installation/user_settings.pm

```diff
@@ -35,12 +35,19 @@ sub run {
     }
 
     $self->enter_userinfo();
-    assert_screen 'inst-userinfostyped';
-    if (match_has_tag 'boo#1122804') {
-        diag "trying again with slow typing due to https://progress.opensuse.org/issues/46190\n";
-        $self->enter_userinfo(retry => 1);
-        assert_screen 'inst-userinfostyped';
+    assert_screen 'inst-userinfostyped-ignore-full-name';
+    for (my $i = 0; $i < 3; $i++) {
+        if (check_screen('inst-userinfostyped-expected-typefaces', 0)) {
+            last;
+        }
+        else {
+            record_info('poo#46190', 'Typing issue with fullname - poo#46190');
+            diag "trying again with slow typing due to https://progress.opensuse.org/issues/46190\n";
+            $self->enter_userinfo(retry => 1);
+            assert_screen 'inst-userinfostyped';
+        }
     }
+    assert_screen('inst-userinfostyped-expected-typefaces');
```

Comment:
so in the case of "all works as expected" and no mistyping you would now have three screen checks and screenshots instead of a single one? I am strongly against that. IMHO code should always be as simple as possible for the happy path and only introducing the additional, more complicated code in case of workaround. I don't think the additional screen checks provide any benefit over the initial `assert_screen` which can cover all alternatives with additional needle tags already.